### PR TITLE
Switch to Server Sent Events based Querying

### PR DESCRIPTION
**Pull request by torbrenner**
This pull request replaces the default Spot implementation with the Rust Spot implementation, which queries the sites using server send envents. Additionally the deployment files are updated to deploy the Rust based spot implementation.

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -20,15 +20,13 @@ services:
 
   spot:
     # NOTE: This will be replaced by https://github.com/samply/spot soon.
-    image: docker.verbis.dkfz.de/ccp-private/central-spot
+    image: samply/rustyspot:refactor-evenMore
```
Comment (Threated) [inline docker-compose.dev.yml:23]: This should be main too right?
```suggestion
    image: samply/rustyspot:main
```

**Review by Threated — CHANGES_REQUESTED** [review]

File: packages/lib/src/classes/spot.ts [hunk 2]
```diff
@@ -36,87 +36,59 @@ export class Spot {
...
-                if (!beamResponses.ok) {
-                    const error: string = await beamResponses.text();
-                    console.debug(
-                        `Received ${beamResponses.status} with message ${error}`,
-                    );
-                    throw new Error(
-                        `Error then retrieving responses from Beam. Abborting requests ...`,
-                    );
-                }
+            console.log(`Created new Beam Task with id ${this.currentTask}`)
```
Comment (patrickskowronekdkfz) [inline packages/lib/src/classes/spot.ts:65]: I would use info instead of log!

File: packages/lib/src/classes/spot.ts [hunk 3]
```diff
@@ -36,87 +36,59 @@ export class Spot {
...
+                    store.set(site, { status: status, data: body })
+                    return store;
+                })
+            })
 
-                    // if the site is already in the store and the status is claimed, don't update the store
-                    if (this.storeCache.get(site)?.status === status) return;
+            // read error events from beam
+            eventSource.addEventListener("error", (message) => {
+                console.log(`Beam returned error ${message}`)
```
Comment (patrickskowronekdkfz) [inline packages/lib/src/classes/spot.ts:83]: I would use error instead of log!

Comment (patrickskowronekdkfz) [inline packages/lib/src/classes/spot.ts:83]: If beam encounters an error, maybe we should inform the user. I'm pretty sure we should do something like this, but not in this PR.

File: packages/lib/src/classes/spot.ts [hunk 4]
```diff
@@ -36,87 +36,59 @@ export class Spot {
...
-                        },
-                    );
-                }
-
-                responseCount = beamResponseData.length;
-                const realResponseCount = beamResponseData.filter(
-                    (response) => response.status !== "claimed",
-                ).length;
+            // event source in javascript throws an error then the event source is closed by backend
+            eventSource.onerror = () => {
```
Comment (patrickskowronekdkfz) [inline packages/lib/src/classes/spot.ts:88]: Maybe add a console.error here

**Review by patrickskowronekdkfz — CHANGES_REQUESTED** [review]
The PR looks good so far, however we should invest time into correct error handling

File: docker-compose.dev.yml (hunk 1 above)
Comment (torbrenner) [reply docker-compose.dev.yml]: change applied

**Review by patrickskowronekdkfz — APPROVED** [review]

File: packages/lib/src/classes/spot.ts (hunk 3 above)
Comment (torbrenner) [reply packages/lib/src/classes/spot.ts]: Yes, i think that is a good idea. Maybe we should open an issue for this. I'think we should just provide an api for the applications to listen for such events, as each application using lens will maybe have their own way of handling it (e.g. showing the notification in the top of the application or as a box in the right bottom corner)

File: packages/lib/src/classes/spot.ts (hunk 4 above)
Comment (torbrenner) [reply packages/lib/src/classes/spot.ts:91]: An error log is not the right choice here. Suprisingly, the eventSource.onerror event is thrown after the backend closes the EventStream, so actually after the beam proxy received answers from all sites. I added an info log instead.

**Review by Threated — APPROVED** [review]

**Pull request merged**